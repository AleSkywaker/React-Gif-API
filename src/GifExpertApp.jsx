import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

const title = 'Gif Expert';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories((categories) => [...categories, newCategory]);
  }

  return (
    <>
      <h1>{title}</h1>
      <AddCategory onNewCategory={onAddCategory} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
