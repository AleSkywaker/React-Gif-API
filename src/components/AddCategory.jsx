import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('one punch');
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input data-testid='shearchGifts' type='text' placeholder='Buscar gifs' value={inputValue} onChange={onInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};