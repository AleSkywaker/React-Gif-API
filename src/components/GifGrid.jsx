
import { GiftItem } from './GiftItem.jsx';
import { useFetchGifts } from '../hooks/useFetchGifts.js';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifts(category);
      return (
        <>
          <h3>{category}</h3>
          {isLoading && (<h2>Cargando...</h2>)}
          <div className='card-grid'>
            {images.map((img) => (
              <GiftItem key={img.id} {...img} />
            ))}
          </div>
        </>
      );
};
