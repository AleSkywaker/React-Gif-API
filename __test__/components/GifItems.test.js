import { render, screen } from '@testing-library/react';
import { GiftItem } from '../../src/components';

describe('Pruebas en <GifItem.jsx/>', () => {
   const title = 'Alex';
   const url = 'http://google.com/';

   test('should match with snapshopt', () => {
      const { container } =render( <GiftItem title={title} url={url}/>)
      expect (container).toMatchSnapshot();
   })

   test('should sohow the url image and the alt attributte', () => {
      render( <GiftItem title={title} url={url}/>)
      // screen.debug();
      const {src, alt} = screen.getByRole('img');
      console.log(url);
      expect( src ).toBe(url)
      expect( alt ).toBe(title)
   })

   test('should show title in component', () => {
    render(<GiftItem title={title} url={url} />);
    expect( screen.getByText(title)).toBeTruthy();
   })

});
