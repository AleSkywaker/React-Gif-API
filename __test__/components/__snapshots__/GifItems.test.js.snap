// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem.jsx/> should match with snapshopt 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Alex"
      src="http://google.com/"
    />
    <p>
      Alex
    </p>
  </div>
</div>
`;
